### DISCUSSION TITLE
`useRouter().currentRoute` is not a ref?

### DISCUSSION THREAD

**Discussion by ProgramRipper** [nuxt/nuxt#24797] [Questions] [answered]
https://stackblitz.com/edit/github-9dpxay?file=app.vue

dependencies:
```
nuxt-app@ /home/projects/github-9dpxay
+-- @esbuild/linux-x64@0.19.9 extraneous
+-- @nuxt/devtools@1.0.6
+-- @parcel/watcher-linux-x64-glibc@2.3.0 extraneous
+-- @parcel/watcher-linux-x64-musl@2.3.0 extraneous
+-- @rollup/rollup-linux-x64-gnu@4.9.0 extraneous
+-- @rollup/rollup-linux-x64-musl@4.9.0 extraneous
+-- node-addon-api@7.0.0 extraneous
+-- nuxt@3.8.2
+-- vue-router@4.2.5
`-- vue@3.3.12
```

The vue-router docs says [Router.currentRoute](https://router.vuejs.org/api/interfaces/Router.html#currentRoute) is a ref of `RouteLocationNormalized`.
However in Nuxt `useRouter().currentRoute` is simply a plain `RouteLocationNormalizedLoaded` object but **not a ref**.

I am new to Nuxt and please teach me if there is something different usage here, thanks!

**Comment by TheAlexLichter** [comment, marked as answer]
You have to use the `pages/` folder to "actually" enable routing. Then, the implementation is also a ref.
See my [StackBlitz](https://stackblitz.com/edit/nuxt-discussions-24797?file=pages%2Findex.vue)

But I agree there is an inconsistency here which we might want to tackle 🤔

**Reply by ProgramRipper** [reply]
Oh, Thanks for your blazing fast answer! Now I have a better knowledge about "Create ./pages/index.vue to enable routing"...

**Reply by TheAlexLichter** [reply]
No problem! Also created an issue to track in https://github.com/nuxt/nuxt/issues/24798
